### DISCUSSION TITLE
Project diamond dependency resolving

### DISCUSSION THREAD

**Discussion by jkawnkwh85** [xmake-io/xmake#3208] [Q&A]
有三个项目，base是基础库，A是在base基础上开发的工具库， B是一个可运行的程序，依赖于base和A。

依赖关系如下

<img width="314" alt="image" src="https://user-images.githubusercontent.com/117454019/209520753-e50225fc-9ea7-42be-8de8-4ed7681edba5.png">

其中B依赖 A ==1.0.4, base ==1.0.8
A(1.0.4版本)依赖 base >= 1.1.0

编译B的时候应该无法找到符合要求的base，但是目前编译的时候不会报错。
请问是我对xmake resolve dependency的方法有误解吗？

**Comment by waruqi** [comment]
？你在做包？

**Reply by jkawnkwh85** [reply]
是的

**Reply by waruqi** [reply]
你 a b 里面 add_deps 怎么配置的？b里面 add_deps base 1.0.8 写死版本了？

**Reply by jkawnkwh85** [reply]
我明白哪里的问题了，我只在项目本身的xmake.lua里写了版本要求，但是没有在xrepo 用的xmake.lua里写版本要求。 感谢！

**Reply by jkawnkwh85** [reply]
在A的 add_deps("base >= 1.1.0"),
B是当前build的项目，没有打包，因此只是写了add_requires("base 1.0.8" "A 1.0.4")。
xmake是不是不会对这种情况做检查的？
